lisa/target.py

```diff
@@ -728,6 +730,30 @@ def remove_kmodule(self, kmodule):
 
         self.target.execute('rmmod {}'.format(kmodule))
 
+    def _init_extend_sched_events(self):
+        """
+        On newer kernels (5.3 and above) PELT signals are present as bare
+        tracepoints and we need to load a module to convert them to proper
+        TRACE_EVENTs.
+
+        This function automatically handles this.
+        """
+
+        sched_events_path = '/sys/kernel/debug/tracing/events/sched/'
+
+        sched_load_avg_task = os.path.join(sched_events_path, 'sched_load_avg_task')
+        sched_load_se = os.path.join(sched_events_path, 'sched_load_se')
+
+        has_sched_load_avg_task = self.target.directory_exists(sched_load_avg_task)
+        has_sched_load_se = self.target.directory_exists(sched_load_se)
+
+        self.need_to_load_sched_tp = not (has_sched_load_avg_task or has_sched_load_se)
+
+        if self.need_to_load_sched_tp:
+            self.build_kmodule('sched_tp')
```

Comment:
Would be worth adding a helper that does all 3 steps in one go, and that `rmmod` first in case the module is already loaded.

Reply:
I was hoping I'd get a nice pointer on how to clean up nicely so that we can't have a left over module loaded when we finish..

_init_extend_sched_events() is a wrapper around these, why create another wrapper?

Reply:
> clean up nicely so that we can't have a left over module loaded when we finish.

There is no cleanup code for Target. We can turn `Target` into a context manager and add some cleanup steps if it's needed, but the code should never rely on the target being cleaned up when re-connecting, since that cleanup may not happen for various reasons.

> _init_extend_sched_events() is a wrapper around these, why create another wrapper?

Seems solved by the generic `_init_kmodules()` in the new version of the PR, which can be used for any module and not just sched stuff.